src/SessionMiddlewareFactory.ts

```diff
@@ -1,37 +1,66 @@
-"use strict";
+import { Request, Response, NextFunction, RequestHandler } from "express";
+import { SessionStore } from "./session/SessionStore";
+import { Failure } from "./error/FailureType";
+import { Either } from "purify-ts";
+import { VerifiedSession } from "./session/model/Session";
+import { SessionHandlerConfig } from "./SessionHandlerConfig";
+import { log } from "./error/ErrorFunctions";
+import expressAsyncHandler from "express-async-handler";
 
-import { RequestHandler } from "express";
-import cookie from "cookie";
-import SessionValidator from "./SessionValidator";
-import SessionStore from "./session/SessionStore";
 
-class SessionMiddlewareFactory {
+export class SessionMiddlewareFactory {
 
-    static create(): RequestHandler {
+    constructor(private readonly config: SessionHandlerConfig,
+        private readonly sessionStore: SessionStore,
+    ) {
 
-        return async function (request, _response, next) {
+        if (!config.cookieSecret) {
+            throw Error("Must provide secret of at least 16 bytes (64 characters) long");
```

Comment:
This error message and one below (at line 31) mismatch. Shall that be 64 characters or 24? Misleading error messages are equally bad as lack of error message.